### PR TITLE
Make download restore backup

### PR DESCRIPTION
**Pull request by prabhakarreddy1234**
### Context

Automate the process of restoring database backup to postgres

### Changes proposed in this pull request

Makefile rule added to start restore process. Downloads backup from azure storage and restores it into postgres.

### Checklist

- [x] [Attach to Trello card](https://trello.com/c/w5bEWodG/400-restore-postgres-backup)
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by merrilmathew** [comment]
[Restore postgres backup](https://trello.com/c/w5bEWodG/400-restore-postgres-backup)

File: Makefile [hunk 1]
```diff
@@ -1,23 +1,42 @@
...
+.PHONY: preprod
 preprod:
 	$(eval DEPLOY_ENV=preprod)
 	$(eval AZURE_SUBSCRIPTION=s165-teachingqualificationsservice-test)
 
+.PHONY: production
 production:
 	$(eval DEPLOY_ENV=production)
 	$(eval AZURE_SUBSCRIPTION=s165-teachingqualificationsservice-production)
+	$(eval AZURE_BACKUP_STORAGE_ACCOUNT_NAME=s165p01dbbackup)
```
Comment (NickGraham101) [inline Makefile:31]: You could use `grep` to get these values from the `<env>.backend.tfvars` files similar to the way that `jq` is used elsewhere in this file to get values from `<env>.tfvars.json` files.  Something like:

`cat terraform/workspace_variables/dev.backend.tfvars | grep -Po "storage_account_name = \"\K\w+"`

Should work and would avoid duplication

File: .github/workflows/database-backup.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
         uses: DFE-Digital/github-actions/install-postgres-client@master
 
       - name: Set environment variable
-        run: echo "BACKUP_FILE_NAME=find-a-lost-trn-production-pg-svc-$(date +"%F-%H-%M-%S")" >> $GITHUB_ENV
+        run: echo "BACKUP_FILE_NAME=find-a-lost-trn-production-pg-svc-$(date +"%F-%H")" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (merrilmathew) [reply .github/workflows/database-backup.yml:56]: So my thinking here was, the developer would know the filename without going to Azure portal and checking in the storage if the daily backups are taken once at night at a particular time. If they have to access the storage account then they might as well download the file for storage?. My vote would be to add --overwrite true. But happy with other option as well.

File: Makefile [hunk 3]
```diff
@@ -92,6 +107,13 @@ restore-postgres: terraform-init read-deployment-config ## make dev restore-post
 	echo "Restoring ${POSTGRES_DATABASE_NAME} from $(TF_VAR_paas_restore_db_from_db_instance) before $(TF_VAR_paas_restore_db_from_point_in_time_before)"
 	make ${DEPLOY_ENV} terraform-apply
 
+restore-data-from-backup: read-deployment-config # make dev restore-data-from-backup CONFIRM_RESTORE=YES
```
Comment (NickGraham101) [inline Makefile:110]: Comment should include `BACKUP_FILENAME="find-a-lost-trn-production-pg-svc-yyyy-mm-dd-HH` as this is a required arg

File: .github/workflows/database-backup.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (NickGraham101) [reply .github/workflows/database-backup.yml:56]: `--overwrite` seems like a good choice since this is the secondary backup method

**Review by saliceti — APPROVED** [review]

**Review by saliceti — APPROVED** [review]

**Pull request merged**